mitiq/lre/inference/multivariate_richardson.py

```diff
@@ -0,0 +1,234 @@
+    variable_exp_no_0_degree = variable_exp_wout_0_degree_list
+
+    # create a list of dict values (just the exponents)
+    variable_exp_list = []
+    for i in variable_exp_no_0_degree:
+        val_i = list(i.values())
+        variable_exp_list.append(val_i)
+
+    for rows, i in enumerate(scale_factor_vectors[1:], start=1):  # type: ignore[assignment]
+        for cols, j in enumerate(variable_exp_list, start=1):
+            evaluated_terms = []
+            for base, exp in zip(list(i), j):
+                # raise scale factor value by the exponent dict value
+                evaluated_terms.append(base**exp)
+            # multiply both elements in the list to create an evaluated
+            # monomial term
+            sample_matrix[rows, cols] = np.prod(evaluated_terms)
+
+    return sample_matrix
+
+
+def linear_combination_coefficients(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+) -> List[np.float64]:
+    r"""
+    Defines the sample matrix required for multivariate extrapolation as
+    defined in :cite:`Russo_2024_LRE`.
+
+    Args:
+        input_circuit: Circuit to be scaled.
+        degree: Degree of the multivariate polynomial.
+        fold_multiplier: Scaling gap required by unitary folding.
+        num_chunks: Number of desired approximately equal chunks. When the
+            number of chunks is the same as the layers in the input circuit,
+            the input circuit is unchanged.
+
+    Returns:
+        List of the evaluated monomial basis terms using the scale factor
+            vectors.
+    """
+    num_layers = len(
+        _get_scale_factor_vectors(
+            input_circuit, degree, fold_multiplier, num_chunks
+        )
+    )
+    input_sample_matrix = sample_matrix(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+    try:
+        det = np.linalg.det(input_sample_matrix)
+    except RuntimeWarning:  # pragma: no cover
+        # taken from https://stackoverflow.com/a/19317237
+        warnings.warn(  # pragma: no cover
+            "To account for overflow error, required determinant of "
+            + "large sample matrix is calculated through "
+            + "`np.linalg.slogdet`."
+        )
+        sign, logdet = np.linalg.slogdet(  # pragma: no cover
+            input_sample_matrix
+        )
+        det = np.exp(logdet)  # pragma: no cover
+
+    if np.isinf(det):
+        raise ValueError(  # pragma: no cover
+            "Determinant of sample matrix cannot be calculated as "
+            + "the matrix is too large. Consider chunking your"
+            + " input circuit. "
+        )
+    assert det != 0.0
+
+    coeff_list = []
+    for i in range(num_layers):
+        sample_matrix_copy = input_sample_matrix.copy()
+        sample_matrix_copy[i] = np.array([[1] + [0] * (num_layers - 1)])
+        coeff_list.append(np.linalg.det(sample_matrix_copy) / det)
```

Comment:
This looks like we are computing the determinant of the minor. If so, can we just delete the values and compute the determinant? Then we can avoid the copying.
```suggestion
        sample_matrix_minor = np.delete(np.delete(sample_matrix, (0), axis=1), (i), axis=0)
        coeff_list.append(np.linalg.det(sample_matrix_minor) / det)
```